types/babel__traverse/index.d.ts

```diff
@@ -11,8 +11,20 @@ import * as t from "@babel/types";
 
 export type Node = t.Node;
 
-export default function traverse<S>(parent: Node | Node[], opts: TraverseOptions<S>, scope: Scope, state: S, parentPath?: NodePath): void;
-export default function traverse(parent: Node | Node[], opts: TraverseOptions, scope?: Scope, state?: any, parentPath?: NodePath): void;
+export default function traverse<S>(
```

Comment:
I'm probably missing something obvious, but why are there two overloads?  Why not just provide a default value for the type parameter of the first overload?

Reply:
I think because then you could do

```ts
traverse<StateType>(node, visitor)
```

In the visitor functions, the type of `state` would appear to be `StateType` even though no state was provided. Ideally, it would error if you specified a type parameter and did not pass in any state.

Reply:
Ah, I missed the extra question mark in the second overload.  Thanks!